cdap-ui/app/wrangler/components/Wrangler/index.js

```diff
@@ -193,18 +212,86 @@ export default class Wrangler extends Component {
     this.setState({wranglerInput: e.target.value});
   }
 
+  renderErrors() {
+    if (this.state.errors.length === 0) {
+      return null;
+    }
+
+    return (
+      <div className="wrangler-error-container">
+        <h4 className="error text-center">Errors:</h4>
+        <table className="table table-bordered error-table">
+          <thead>
+            <tr>
+              <th>Row</th>
+              <th>Error</th>
+            </tr>
+          </thead>
+          <tbody>
+            {
+              this.state.errors.map((error) => {
+                return (
+                  <tr key={shortid.generate()}>
+                    <td>{error.row}</td>
+                    <td>{error.message}</td>
+                  </tr>
+                );
+              })
+            }
+          </tbody>
+        </table>
+      </div>
+    );
+
+  }
+
+  preventPropagation (e) {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
+  }
+
   renderWranglerCopyPaste() {
-    if (!this.state.textarea) {
+    if (!this.state.textarea || this.state.file.name) {
       return (
         <div
           className="wrangler-plus-button text-center"
           onClick={this.onPlusButtonClick}
         >
-          <div className="plus-button">
-            <i className="fa fa-plus-circle"></i>
+          <div
+            className="dropzone-container"
+            onClick={(e) => this.preventPropagation(e)}
+          >
+            <Dropzone
```

Comment:
It is not consistent. Either we need to then change the way we have it in Stream ingest fast action or the FileDnD needs to be extended to show the file browser on click. Or `<EditableFileDnD />` may be?